console-subscriber/src/builder.rs

```diff
@@ -101,4 +104,41 @@ impl Builder {
     pub fn build(self) -> (TasksLayer, Server) {
         TasksLayer::build(self)
     }
+
+    /// Configures this builder from a standard set of environment variables:
+    ///
+    /// * `TOKIO_CONSOLE_RETENTION`: The number of seconds to accumulate
+    ///   completed tracing data. Default: 60s.
+    /// * `TOKIO_CONSOLE_BIND`: a HOST:PORT description, such as
+    ///   localhost:1234 or similar. Default: 127.0.0.1:6669
+    /// * `TOKIO_CONSOLE_PUBLISH_INTERVAL`: The number of milliseconds to wait
+    ///   between sending updates to the console. Default: 1000ms (1s)
+
```

Comment:
take it or leave it: it would be nice if the docs for this also included an example of using the builder to configure a console layer from the default environment variables , and then using it to build a subscriber, spawn the background task, etc? but, we can come back and add that later.

Reply:
Before I saw your review on this PR, I attempted to address this use case by adding a single in-code extension point: `init_with_layer`, with the hope that we can accommodate all common initialization patterns and discourage anyone who would need docs on how to eject from doing so. With the addition of init_with_layer, have we covered all of those use cases? An advantage of this approach is that if the majority of users are going through `init` or `init_with_layer`, future changes to those functions wouldn't require app code change, whereas if we encourage a copy-paste initializer through documentation, we have no way of pushing out changes to that "ejected" code.